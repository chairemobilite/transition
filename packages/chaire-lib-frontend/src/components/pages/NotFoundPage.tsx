/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';
import { Link } from 'react-router';
import { useTranslation } from 'react-i18next';

export const NotFoundPage: React.FC = () => {
    const { t } = useTranslation('auth');

    return (
        <div>
            404 - <Link to="/">{t('auth:BackToHomePage')}</Link>
        </div>
    );
};

export default NotFoundPage;
