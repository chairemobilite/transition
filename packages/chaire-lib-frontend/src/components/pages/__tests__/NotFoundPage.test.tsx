/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import NotFoundPage from '../NotFoundPage';

test('Should correctly render NotFound page', () =>{
    const wrapper = mount(<BrowserRouter><NotFoundPage /></BrowserRouter>);
    expect(wrapper).toMatchSnapshot();
});
