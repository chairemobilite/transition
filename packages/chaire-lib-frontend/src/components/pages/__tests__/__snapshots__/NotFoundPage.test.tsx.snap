// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render NotFound page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withI18nextTranslation(NotFoundPage)>
      <NotFoundPage
        i18n={Object {}}
        t={[Function]}
        tReady={false}
      >
        <div>
          404 - 
          <Link
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                BackToHomePage
              </a>
            </LinkAnchor>
          </Link>
        </div>
      </NotFoundPage>
    </withI18nextTranslation(NotFoundPage)>
  </Router>
</BrowserRouter>
`;
